pkg/reconciler/v1alpha1/pipelinerun/validate.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipelinerun
+
+import (
+	"fmt"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+)
+
+// validate all ref exist
+func (c *Reconciler) validate(pr *v1alpha1.PipelineRun) (*v1alpha1.Pipeline, string, error) {
```

Comment:
I think it would be better to extract the `validate` out  and not make it part of the reconciler, its not really a behavior  of the reconciler. Maybe you can pass the reconciler as an input to get all the listers

Reply:
Thats a good point. Will update the PR